haystack/modeling/training/base.py

```diff
@@ -371,7 +373,24 @@ def train(self):
 
     def compute_loss(self, batch: dict, step: int) -> torch.Tensor:
         # Forward & backward pass through model
-        logits = self.model.forward(**batch)
+        if isinstance(self.model, AdaptiveModel):
+            logits = self.model.forward(
+                input_ids=batch["input_ids"], segment_ids=None, padding_mask=batch["padding_mask"]
+            )
+
+        elif isinstance(self.model, BiAdaptiveModel):
+            logits = self.model.forward(
+                query_input_ids=batch["query_input_ids"],
+                query_segment_ids=batch["query_segment_ids"],
+                query_attention_mask=batch["query_attention_mask"],
+                passage_input_ids=batch["passage_input_ids"],
+                passage_segment_ids=batch["passage_segment_ids"],
+                passage_attention_mask=batch["passage_attention_mask"],
+            )
+
```

Comment:
We might need to support `TriAdaptiveModel` as well?

Reply:
Yes, we are indirectly as `TriAdaptiveModel ` is using kwargs